sap-2-tier-marketplace-image/shared/os-disk-parts.json

```diff
@@ -100,7 +101,7 @@
             "uri": "[parameters('userImageVhdUri')]"
           },
           "vhd": {
-            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', resourceGroup().name, '/', parameters('vmName'), '-', variables('osDiskName'))]"
+            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', variables('resourceGroupLower'), '/', parameters('vmName'), '-', variables('osDiskName'))]"
```

Comment:
Need to make sure there are no underscores either - see: https://azure.microsoft.com/en-us/documentation/articles/guidance-naming-conventions/#naming-rules-and-restrictions

Which is why you don't see this pattern used for storage resources...